js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -58,70 +61,102 @@ export const LaunchAssetExecutionButton: React.FC<{
       disabledReason || 'Assets must be in the same job to be materialized together.';
   }
 
+  const assetsWithConfig = assets.filter((a) => configSchemaForAssetNode(a));
+  const anyAssetsHaveConfig = assetsWithConfig.length > 0;
+
+  if (anyAssetsHaveConfig && partitionDefinition) {
+    disabledReason =
+      disabledReason || 'Cannot materialize assets using both asset config and partitions';
+  }
+
   title = title || 'Refresh';
-  if (partitionDefinition) {
+
+  let tooltipChildren: React.ReactNode;
+  if (anyAssetsHaveConfig) {
+    const assetOpNames = assets.map(({opNames}) => (opNames as string[])[0]);
```

Comment:
I think this will crash if the user ever manages to get a "source asset" into their selection. It shouldn't happen, but it'd be safer to do this I think:

```
    const assetOpNames = assets.map(({opNames}) => (opNames?.[0]).filter(Boolean);
```

I'm not sure if this is intentionally taking the first item (I guess only single-op assets can be configured maybe?) But it'd also be clean to do this and gracefully handle cases where that array is not a single item:

```
    const assetOpNames = assets.flatMap((a) => a.opNames || []);
```

Reply:
Good call, fixed (to `flatten` version).